src/cli.rs

```diff
@@ -110,3 +165,80 @@ impl TryFrom<&'_ clap::ArgMatches<'static>> for Command {
     }
 }
 
+
+impl TryFrom<&'_ clap::ArgMatches<'static>> for WgCommand {
+    type Error = Error;
+    fn try_from(args: &clap::ArgMatches<'static>) -> Result<Self> {
+
+
+            if let Some(wg_command) = args.subcommand_matches("wireguard")
+            {
+                // Unlock
+                if let Some (command) = wg_command.subcommand_matches("unlock")
+                {
+                        let pin = match command.value_of("PIN") {
+                            Some(s) => {s},
+                            None => {return Err(anyhow::anyhow!("Could not parse pin"));}
+    
+                        };
+                       return Ok(WgCommand::Unlock(Unlock {
```

Comment:
Just stylistic, but similar to the `hex_string_to32_char_array` below, combinators compactify this kind of thing:

```
let pin = command.value_of("PIN").ok_or_else(|| anyhow::anyhow!("Could not parse pin"))?.to_string();
return Ok(WgCommand::Unlock(Unlock { pin }))
```

Note that `return` is usually only used for explicit early error returns, if you can setup the code that the main Result of the function is "at then end", you can leave out the `return` keyword.